packages/node-experimental/src/sdk/hub.ts

```diff
@@ -0,0 +1,170 @@
+import type {
+  Client,
+  CustomSamplingContext,
+  EventHint,
+  Hub,
+  Integration,
+  IntegrationClass,
+  Session,
+  Severity,
+  SeverityLevel,
+  TransactionContext,
+} from '@sentry/types';
+
+import {
+  addBreadcrumb,
+  captureEvent,
+  captureException,
+  captureMessage,
+  configureScope,
+  endSession,
+  getClient,
+  getCurrentScope,
+  lastEventId,
+  setContext,
+  setExtra,
+  setExtras,
+  setTag,
+  setTags,
+  setUser,
+  startSession,
+  withScope,
+} from './api';
+import { callExtensionMethod, getGlobalCarrier } from './globals';
+import type { Scope } from './scope';
+import type { SentryCarrier } from './types';
+
+/** Ensure the global hub is our proxied hub. */
+export function setupGlobalHub(): void {
+  const carrier = getGlobalCarrier();
+  carrier.hub = getCurrentHub();
+}
+
+/**
+ * This is for legacy reasons, and returns a proxy object instead of a hub to be used.
+ */
+export function getCurrentHub(): Hub {
```

Comment:
Should we deprecate this?

Reply:
I mean the whole idea of the scope changes is to get rid of `getCurrentHub`, so I agree 😅